### ISSUE TITLE
TemplateDelims can eat an entire range operator

### ISSUE THREAD

**Issue by JRaspass** [tdewolff/minify#698]
```go
package main

import (
	"fmt"

	"github.com/tdewolff/minify/v2"
	"github.com/tdewolff/minify/v2/html"
)

func main() {
	data := `<select>
	{{ range . }}
		<option>{{ . }}
	{{ end }}
	</select>`

	fmt.Println(data)
	fmt.Println("---")

	m := minify.New()
	m.Add("text/html", &html.Minifier{TemplateDelims: html.GoTemplateDelims})

	var err error
	if data, err = m.String("text/html", data); err != nil {
		panic(err)
	}

	fmt.Println(data)
}
```
https://go.dev/play/p/YkV7UfwvmIk
```html
<select>
	{{ range . }}
		<option>{{ . }}
	{{ end }}
	</select>
```
```html
<select><option>{{ . }}
	{{ end }}
	</select>
```

Not all range operators in my templates were eaten but something about the way this one is written really confuses minify, I've tried to golf it down a bit.

**Comment by tdewolff** [comment]
Thanks James, this was due to text being stripped out of `<select>` elements, should be fixed now!

**Comment by tbo** [comment]
@tdewolff I'm experiencing a similar issue with the latest version, v2.20.37:

```
      <select id="variant-selector">
        {{- range $variant := $product.variants -}}
        <option value="{{$variant.sku}}">{{$variant.abbreviated_size}}</option>
        {{- end -}}
      </select>
```

Becomes

```
<select id=variant-selector>
        {{- range $variant := $product.variants -}}
        <option value="{{$variant.sku}}">{{$variant.abbreviated_size}}</select>
```

The "{{- end -}}" is being removed for some reason. I tried all the "Keep" options, but the outcome remains the same. Reproduction: https://go.dev/play/p/yYnKbuYz6vA

**Comment by tdewolff** [comment]
Should be fixed in v2.21.0, please let me know if you run into trouble.

**Comment by tbo** [comment]
@tdewolff Works like a charm!
